lab_4_fill_words_by_ngrams/main.py

```diff
@@ -220,6 +294,25 @@ def _calculate_perplexity(self, generated_text: str) -> float:  # type: ignore
                 or if methods used return None,
                 or if nothing was generated.
         """
+        if not (isinstance(generated_text, str) and generated_text):
+            raise ValueError
+        encoded = self._word_processor.encode(generated_text)
+        if not encoded:
+            raise ValueError
+        n_gram_size = self._language_model.get_n_gram_size()
+        log_sum = 0.0
+        for ind in range(n_gram_size - 1, len(encoded)):
+            context = tuple(encoded[ind - n_gram_size + 1: ind])
```

Comment:
is it necessary?

Reply:
solved, it was just for me not to get confused or forget something)